internal/controller/serverclaim_controller.go

```diff
@@ -385,6 +385,10 @@ func (r *ServerClaimReconciler) claimServerBySelector(ctx context.Context, log l
 			log.V(1).Info("Server not in a claimable state", "Server", s.Name, "ServerState", s.Status.State)
 			continue
 		}
+		if s.Status.PowerState != metalv1alpha1.ServerOffPowerState {
```

Comment:
Shouldn't we handling this in `claimServerByReference` as well?

Reply:
added it there, too